### PR TITLE
Bump version and CHANGELOG for v3.42.0

### PR DESCRIPTION
**Pull request by mitchbne**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -898,7 +926,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - Experimental `--spawn` option to spawn multiple parallel agents [#590](https://github.com/buildkite/agent/pull/590) (@lox) - **Update:** This feature is now super stable.
 - Add a linux/ppc64le build target [#859](https://github.com/buildkite/agent/pull/859) (@lox)
-- Basic metrics collection for Datadog [#832](https://github.com/buildkite/agent/pull/832) (@lox)
+- Basic metrics collection for baseVersion [#832](https://github.com/buildkite/agent/pull/832) (@lox)
```
Comment (triarius) [inline CHANGELOG.md:929]: Was this intentional?

Comment (DrJosh9000) [inline CHANGELOG.md:929]: Unintended change here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,34 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [v3.41.0](https://github.com/buildkite/agent/tree/v3.42.0) (2023-01-05)
+[Full Changelog](https://github.com/buildkite/agent/compare/v3.41.0...v3.42.0)
+
+ ### Added
+ - Add `git config protocol.file.allow=always` to tests [#1895](https://github.com/buildkite/agent/pull/1895) (@triarius)
+ - Add an in-built hierarchical status page [#1873](https://github.com/buildkite/agent/pull/1873) (@DrJosh9000)
+ - Add an `agent-startup` hook that fires at the same time as the `agent-shutdown` hook is registered [#1778](https://github.com/buildkite/agent/pull/1778) (@donalmacc)
+ - Add `test-collector` plugin for internal metric collection [#1881](https://github.com/buildkite/agent/pull/1881) (@DrJosh9000)
```
Comment (DrJosh9000) [inline CHANGELOG.md:15]: "internal metric collection" sounds wrong? But the PR doesn't add anything for users, only us, so might not be worth mentioning.

**Review by triarius — COMMENTED** [review]
Thanks for figuring out how to generate this @mitchbne. It's a pretty tedious process that you can now give feed back on.

Usually, we don't include test and CI fixes in the change log, and it looks like a lot of these are in that category, so I think we should remove or maybe summarise them with something like:
```
# Fixed
- lots of improvements to tests and CI (...)
```
where the `(...)` is replaced with a list of the authors of the PRs that were squashed into this changelog item.

**Comment by DrJosh9000** [comment]
@triarius Haha, jinx

File: CHANGELOG.md (hunk 1 above)
Comment (mitchbne) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (mitchbne) [reply CHANGELOG.md]: Removed from changelog :)

File: CHANGELOG.md (hunk 1 above)
Comment (mitchbne) [reply CHANGELOG.md]: Nope :(

**Comment by mitchbne** [comment]
Thanks for the reviews; I've addressed the comments. Ready for a re-review :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,28 @@ All notable changes to this project will be documented in this file.
...
 
+## [v3.41.0](https://github.com/buildkite/agent/tree/v3.42.0) (2023-01-05)
+[Full Changelog](https://github.com/buildkite/agent/compare/v3.41.0...v3.42.0)
+
+ ### Added
+ - Add an in-built hierarchical status page [#1873](https://github.com/buildkite/agent/pull/1873) (@DrJosh9000)
+ - Add an `agent-startup` hook that fires at the same time as the `agent-shutdown` hook is registered [#1778](https://github.com/buildkite/agent/pull/1778) (@donalmacc)
+
+ ### Changed
+- Use `stdlib` errors instead of `github.com/pkg/errors` [#1872](https://github.com/buildkite/agent/pull/1872) (@DrJosh9000)
```
Comment (DrJosh9000) [inline CHANGELOG.md:16]: - Feels a bit weird to put stdlib in code quotes when it's not a specific package / module / ... `errors` is a package though.
- It's also an internal cleanup, so is it worth mentioning at all?

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,28 @@ All notable changes to this project will be documented in this file.
...
+- Update `dependabot.yml` to include packaging Dockerfiles [#1861](https://github.com/buildkite/agent/pull/1861) (@triarius)
+- Enforce a timeout on `finishJob` and `onUploadChunk` [#1854](https://github.com/buildkite/agent/pull/1854) (@DrJosh9000)
+- A variety of dependency updates, documentation, and code cleanups! (@dependabot[bot]) (@DrJosh9000) (@moskyb)
+- Flakey test fixes and test suite enhandcements (@triarius) (@DrJosh9000)
+
+ ### Fixed
+ - Use `pkill` instead of `ps-grep-awk-kill` when restarting agents after upgrade [#1875](https://github.com/buildkite/agent/pull/1875) (@DrJosh9000)
+ - Ensure that unrecoverable errors for Hearbeat and Ping stop the agent [#1855](https://github.com/buildkite/agent/pull/1855) (@moskyb)
```
Comment (DrJosh9000) [inline CHANGELOG.md:24]:
```suggestion
 - Ensure that unrecoverable errors for Heartbeat and Ping stop the agent [#1855](https://github.com/buildkite/agent/pull/1855) (@moskyb)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,28 @@ All notable changes to this project will be documented in this file.
...
+ ### Added
+ - Add an in-built hierarchical status page [#1873](https://github.com/buildkite/agent/pull/1873) (@DrJosh9000)
+ - Add an `agent-startup` hook that fires at the same time as the `agent-shutdown` hook is registered [#1778](https://github.com/buildkite/agent/pull/1778) (@donalmacc)
+
+ ### Changed
+- Use `stdlib` errors instead of `github.com/pkg/errors` [#1872](https://github.com/buildkite/agent/pull/1872) (@DrJosh9000)
+- Update `dependabot.yml` to include packaging Dockerfiles [#1861](https://github.com/buildkite/agent/pull/1861) (@triarius)
+- Enforce a timeout on `finishJob` and `onUploadChunk` [#1854](https://github.com/buildkite/agent/pull/1854) (@DrJosh9000)
+- A variety of dependency updates, documentation, and code cleanups! (@dependabot[bot]) (@DrJosh9000) (@moskyb)
+- Flakey test fixes and test suite enhandcements (@triarius) (@DrJosh9000)
```
Comment (DrJosh9000) [inline CHANGELOG.md:20]:
```suggestion
- Flakey test fixes and test suite enhancements (@triarius, @DrJosh9000)
```

**Review by DrJosh9000 — COMMENTED** [review]
Couple of typos but looking good!

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,28 @@ All notable changes to this project will be documented in this file.
...
+
+ ### Fixed
+ - Use `pkill` instead of `ps-grep-awk-kill` when restarting agents after upgrade [#1875](https://github.com/buildkite/agent/pull/1875) (@DrJosh9000)
```
[the remark this answers is not in the document]
Comment (mitchbne) [reply CHANGELOG.md]: Removed from the list

**Review by triarius — APPROVED** [review]
LGTM, nice work!

**Review by DrJosh9000 — APPROVED** [review]
LGTM 🥳

**Pull request merged**